apps/gitness/src/pages/run-pipeline-dialog/run-pipeline-dialog.tsx

```diff
@@ -1,167 +1,21 @@
-      path: `${ADDITIONAL_INPUTS_PREFIX}.branch`,
-      inputConfig: {
-        options: listBranchesData?.map(branchItem => ({ label: branchItem?.name, value: branchItem?.name }))
-      }
-    },
-    { inputType: InputType.separator, path: '' }
-  ]
-  formDefinition.inputs = [...additionalInput, ...formDefinition.inputs]
-
-  const resolver = useZodValidationResolver(formDefinition)
-
-  const { mutateAsync: createExecutionAsync, isLoading: isLoadingCreateExecution } = useCreateExecutionMutation({})
-
-  const runPipeline = ({ branch, inputsValues }: { branch: string; inputsValues: Inputs }) => {
-    createExecutionAsync({
-      pipeline_identifier: pipelineId ?? '',
-      repo_ref: repoRef,
-      queryParams: { branch: branch },
-      body: inputsValues
-    })
-      .then(response => {
-        onClose()
-
-        const executionId = response.number
-        navigate(`${toExecutions}/${executionId}`)
-        // TODO: toast here ?
-      })
-      .catch(error => {
-        console.error(error)
-        // TODO: error toast here ?
-      })
-  }
-  if (loading || listBranchesLoading) {
-    // TODO
-    return <>'Loading...'</>
-  }
-
+export default function RunPipelineDialog({ onClose, open, ...rest }: RunPipelineDialogProps) {
   return (
-    <RootForm
-      defaultValues={{ _: { branch } }}
-      resolver={resolver}
-      mode="onSubmit"
-      onSubmit={values => {
-        const transformers = getTransformers(formDefinition)
-        const transformedValues = outputTransformValues(values, transformers)
-
-        const branch = transformedValues._.branch
-        const inputsValues = omit(transformedValues, '_')
-
-        runPipeline({ branch, inputsValues })
-      }}
-      validateAfterFirstSubmit={true}>
-      {rootForm => (
-        <Dialog
-          open={open}
-          onOpenChange={open => {
-            if (!open) onClose()
-          }}>
-          <DialogContent className="max-w-[500px] bg-primary-background border-border">
-            <DialogHeader>
-              <DialogTitle>Run Pipeline</DialogTitle>
-            </DialogHeader>
-            <DialogDescription>
-              <Spacer size={6} />
-              <RenderForm className="space-y-4" factory={inputComponentFactory} inputs={formDefinition} />
-            </DialogDescription>
-            <DialogFooter>
-              <Button onClick={onClose} className="text-primary" variant="outline">
-                Cancel
-              </Button>
-              <Button loading={isLoadingCreateExecution} onClick={() => rootForm.submitForm()}>
-                Run pipeline
-              </Button>
-            </DialogFooter>
-          </DialogContent>
-        </Dialog>
-      )}
-    </RootForm>
+    <Dialog
+      open={open}
+      onOpenChange={open => {
+        if (!open) onClose()
+      }}>
+      <DialogContent className="max-w-[500px] bg-primary-background border-border">
+        <RunPipelineForm {...rest} requestClose={() => onClose()} />
```

Comment:
quick question: why do we use {..rest} here

Reply:
We are destructuring props `...{ onClose, open, ...rest }: RunPipelineDialogProps...`
so `rest` contains all props except `onClose, open`. If you look at the `RunPipelineDialogProps` type you will see that it extends `RunPipelineFormProps`. This means that  `rest` contains all props defined in the `RunPipelineFormProps` type.